scripts/zkai_d128_cryptographic_backend_gate.py

```
@@ -0,0 +1,1176 @@
+    if not isinstance(payload, dict):
+        raise D128CryptographicBackendGateError(f"source evidence must be a JSON object: {path}", layer="source_proof_native_contract")
+    return payload
+
+
+@functools.lru_cache(maxsize=4)
+def load_checked_proof_native(path: pathlib.Path = PROOF_NATIVE_EVIDENCE) -> dict[str, Any]:
+    payload = load_json(path)
+    try:
+        PROOF_NATIVE.validate_payload(payload)
+    except Exception as err:  # noqa: BLE001 - normalize imported validator errors.
+        raise D128CryptographicBackendGateError(f"proof-native source validation failed: {err}", layer="source_proof_native_contract") from err
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)